did_key/src/consumer.rs

```diff
@@ -32,19 +32,57 @@ async fn resolve_did(did: &str) -> std::result::Result<CoreDocument, Box<dyn std
+    use test_log::test;
+
+    #[test(tokio::test)]
     async fn resolves_did_key() {
         let did = "did:key:z6Mkk7yqnGF3YwTrLpqrW6PGsKci7dNqh1CjnvMbzrMerSeL";
         let document = resolve_did(did).await.unwrap();
 
         assert_eq!(
-            document.id(),
-            "did:key:z6Mkk7yqnGF3YwTrLpqrW6PGsKci7dNqh1CjnvMbzrMerSeL"
-        );
-
-        assert_eq!(
-            document.verification_method().first().unwrap().id().to_string(),
-            "did:key:z6Mkk7yqnGF3YwTrLpqrW6PGsKci7dNqh1CjnvMbzrMerSeL#z6Mkk7yqnGF3YwTrLpqrW6PGsKci7dNqh1CjnvMbzrMerSeL"
+            document.to_json_value().unwrap(),
+            json!({
+                "@context": "https://www.w3.org/ns/did/v1", // TODO: <== not according to spec! (should be array)
+                "id": "did:key:z6Mkk7yqnGF3YwTrLpqrW6PGsKci7dNqh1CjnvMbzrMerSeL",
+                "verificationMethod": [
+                  {
+                    "id": "did:key:z6Mkk7yqnGF3YwTrLpqrW6PGsKci7dNqh1CjnvMbzrMerSeL#z6Mkk7yqnGF3YwTrLpqrW6PGsKci7dNqh1CjnvMbzrMerSeL",
+                    "type": "JsonWebKey2020",
+                    "controller": "did:key:z6Mkk7yqnGF3YwTrLpqrW6PGsKci7dNqh1CjnvMbzrMerSeL",
+                    "publicKeyJwk": {
+                      "kty": "OKP",
+                      "crv": "Ed25519",
+                      "x": "VDXDwuGKVq91zxU6q7__jLDUq8_C5cuxECgd-1feFTE"
+                    }
+                  },
+                  {
+                    "id": "did:key:z6Mkk7yqnGF3YwTrLpqrW6PGsKci7dNqh1CjnvMbzrMerSeL#z6LSrdqo4M24WRDJj1h2hXxgtDTyzjjKCiyapYVgrhwZAySn",
+                    "type": "OKP", // TODO: <== not according to spec! (should be "JsonWebKey2020")
+                    "controller": "did:key:z6Mkk7yqnGF3YwTrLpqrW6PGsKci7dNqh1CjnvMbzrMerSeL",
+                    "publicKeyJwk": {
+                      "kty": "OKP",
+                      "crv": "X25519",
+                      "x": "3kY9jl1by7pLzgJktUH-e9H6fihdVUb00-sTzkfmIl8"
+                    }
+                  }
+                ],
+                "authentication": [
+                  "did:key:z6Mkk7yqnGF3YwTrLpqrW6PGsKci7dNqh1CjnvMbzrMerSeL#z6Mkk7yqnGF3YwTrLpqrW6PGsKci7dNqh1CjnvMbzrMerSeL"
+                ],
+                "assertionMethod": [
+                  "did:key:z6Mkk7yqnGF3YwTrLpqrW6PGsKci7dNqh1CjnvMbzrMerSeL#z6Mkk7yqnGF3YwTrLpqrW6PGsKci7dNqh1CjnvMbzrMerSeL"
+                ],
+                "capabilityDelegation": [
+                  "did:key:z6Mkk7yqnGF3YwTrLpqrW6PGsKci7dNqh1CjnvMbzrMerSeL#z6Mkk7yqnGF3YwTrLpqrW6PGsKci7dNqh1CjnvMbzrMerSeL"
+                ],
+                "capabilityInvocation": [
+                  "did:key:z6Mkk7yqnGF3YwTrLpqrW6PGsKci7dNqh1CjnvMbzrMerSeL#z6Mkk7yqnGF3YwTrLpqrW6PGsKci7dNqh1CjnvMbzrMerSeL"
+                ],
+                "keyAgreement": [
+                  "did:key:z6Mkk7yqnGF3YwTrLpqrW6PGsKci7dNqh1CjnvMbzrMerSeL#z6LSrdqo4M24WRDJj1h2hXxgtDTyzjjKCiyapYVgrhwZAySn"
+                ]
+            })
```

Comment:
`[resolved]` 
Does this mean that `identity.rs` is producing invalid documents?

Reply:
No, we're using the `did-key` crate here to generate the resolved DID document (which is not up-to-date with the standard). There was no need to patch it manually with identity.rs until now, but we might need to if there's problems on the consumer side.